### PR TITLE
Make sure the default User-Agent matches the spec

### LINKED ISSUE #30: Make sure the default User-Agent matches the spec
See: http://tools.ietf.org/html/rfc2616#section-14.43

### PR DESCRIPTION
**Pull request by khoomeister**
fixes #30

### CODE REVIEW

**Comment by khoomeister** [comment]
Just thought I'll continue the conversation here :smiley:

File: test/helpers.test.coffee [hunk 1]
```diff
@@ -34,7 +34,7 @@ describe 'Helpers', ->
   describe "buildUserAgent", ->
     it 'has defaults', ->
       ua = buildUserAgent({})
-      assert.equal "[no serviceName]/[no serviceVersion]; [no appName]/[no appSha]; [no fqdn]", ua
+      assert.equal "[no serviceName]/[no serviceVersion] ([no appName]/[no appSha]; [no fqdn])", ua
```
Comment (jkrems) [inline test/helpers.test.coffee]: `[` is not a valid token:

```
       token          = 1*<any CHAR except CTLs or separators>
       separators     = "(" | ")" | "<" | ">" | "@"
                      | "," | ";" | ":" | "\" | <">
                      | "/" | "[" | "]" | "?" | "="
                      | "{" | "}" | SP | HT
```

Maybe just `noServiceName/noServiceVersion`?

File: test/request.test.coffee [hunk 2]
```diff
@@ -117,8 +116,8 @@ describe 'actually making a request', ->
       assert.equal 'sneaky header', reqMirror.headers['x-sneaky']
 
       userAgent = reqMirror.headers['user-agent']
-      assert.include 'myApi/1.0.0', userAgent
-      assert.include 'myApp/mySha', userAgent
+      console.log userAgent
```
Comment (jkrems) [inline test/request.test.coffee]: I'm guessing this should not be here.

**Comment by jkrems** [comment]
Looks good in general!

File: test/helpers.test.coffee (hunk 1 above)
Comment (khoomeister) [inline test/helpers.test.coffee]: Sounds good - 'tis done!

**Comment by jkrems** [comment]
:+1:

**Comment by khoomeister** [comment]
v3.0.0

**Comment by jkrems** [comment]
Are we sure about `v3.0.0`? It's arguably a bug that we used a non-compliant user agent before.

**Comment by khoomeister** [comment]
OK backpeddling - v3.0.0 is unpublished - 'tis now v2.3.1.

**Pull request merged**